### DISCUSSION TITLE
Unable to use kubectl cnpg plugin

### DISCUSSION THREAD

**Discussion by arashnikoo** [cloudnative-pg/cloudnative-pg#229] [Q&A] [answered]
I am unable to use the plugin due to this error:

```
$ kubectl cnpg version
Error: exec plugin: invalid apiVersion "client.authentication.k8s.io/v1alpha1"
```
Here's some additional info

```
$ kubectl version --output=yaml
clientVersion:
  buildDate: "2022-04-14T08:49:13Z"
  compiler: gc
  gitCommit: ad3338546da947756e8a88aa6822e9c11e7eac22
  gitTreeState: clean
  gitVersion: v1.23.6
  goVersion: go1.17.9
  major: "1"
  minor: "23"
  platform: darwin/amd64
serverVersion:
  buildDate: "2022-04-01T03:14:50Z"
  compiler: gc
  gitCommit: 09ff856a7a3e49ba29b28b5532cde35fdac5ec60
  gitTreeState: clean
  gitVersion: v1.20.15-eks-14c7a48
  goVersion: go1.15.15
  major: "1"
  minor: 20+
  platform: linux/amd64
```

I tried upgrade and downgrade `kubectl` version but no luck so far.

**Comment by litaocdl** [comment]
May you please check if the context in kube config for target cluster contains "client.authentication.k8s.io/v1alpha1", and if that could be upgrade ?

**Comment by arashnikoo** [comment]
The config does contain that:

```
    exec:
      apiVersion: client.authentication.k8s.io/v1alpha1
```

Unfortunately, I don't have access to the cluster to upgrade it.

**Comment by litaocdl** [comment]
It seems to be the client you used for authentication need upgrade, which tools are u using for authentication to cluster ? teleport ?

**Comment by sxd** [comment]
Hi @arashnikoo 

I'll convert this into a discussion, since doesn't seem to be an issue with the plugin but an issue with the libs installed in the current environment.

Regards!

**Comment by phisco** [comment, marked as answer]
From [here](https://github.com/fluxcd/flux2/issues/2817) it looks like it would be enough to change the kubeconfig to `client.authentication.k8s.io/v1beta1` instead of `client.authentication.k8s.io/v1alpha1`.

**Reply by gbartolini** [reply]
Given the lack of responsiveness, I have marked this as the answer - in any case the issue does not seem to be related to the plugin.

**Comment by gbartolini** [comment]
@arashnikoo were you able to solve your issue?
